internal/util/util_test.go

```diff
@@ -0,0 +1,130 @@
+package util
+
+import (
+	"bytes"
+	"context"
+	"encoding/base64"
+	"reflect"
+	"testing"
+
+	"cloud.google.com/go/compute/metadata"
+	"github.com/containerd/containerd/namespaces"
+	"github.com/google/go-tpm-tools/client"
+	"github.com/google/go-tpm-tools/internal/test"
+	"github.com/google/go-tpm-tools/proto/attest"
+	"github.com/google/go-tpm-tools/verifier"
+	"golang.org/x/oauth2"
+	"golang.org/x/oauth2/google"
+)
+
+func TestPrincipleFetcher(t *testing.T) {
+	var dummyMetaInstance = Instance{ProjectID: "test-project", ProjectNumber: "1922337278274", Zone: "us-central-1a", InstanceID: "12345678", InstanceName: "default"}
+	mockMdsServer, err := NewMetadataServer(dummyMetaInstance)
+	if err != nil {
+		t.Error(err)
+	}
+	defer mockMdsServer.Stop()
+	mdsClient := metadata.NewClient(nil)
+	gotTokens, err := PrincipalFetcher("test_audience", mdsClient)
+	if err != nil {
+		t.Error(err)
+	}
+	wantTokens := [][]byte{[]byte("test_jwt_token")}
+	if !reflect.DeepEqual(wantTokens, gotTokens) {
```

Comment:
go/go-style/decisions#types-of-equality here and below `cmp.Equal` is more preferable.